paasta_tools/cli/cmds/spark_run.py

```diff
@@ -1182,6 +1209,10 @@ def paasta_spark_run(args):
         needs_docker_cfg=needs_docker_cfg,
         aws_region=args.aws_region,
         force_spark_resource_configs=args.force_spark_resource_configs,
+        use_eks=args.use_eks,
+        k8s_server_address=_get_k8s_url_for_cluster(args.cluster)
+        if args.use_eks
```

Comment:
Nit: I would rather prefer this if condition earlier in the code as opposed to inside the function parameters.

Something like:
def _get_k8s_url_for_cluster(cluster: str, use_eks: bool) -> Optional[str]:
    if use_eks: return xyz
    else return None